tests/test_cache.py

```diff
@@ -0,0 +1,96 @@
+from contextlib import contextmanager
+from os import remove
+from shutil import rmtree
+from tempfile import NamedTemporaryFile, mkdtemp
+
+from pytest import raises
+
+from piptools.cache import read_cache_file, CorruptCacheError, DependencyCache
+
+
+@contextmanager
+def _read_cache_file_helper(to_write):
+    """
+    On enter, create the file with the given string, and then yield its path.
+    On exit, delete that file.
+
+    :param str to_write: the content to write to the file
+    :yield: the path to the temporary file
+    """
+    # Create the file and write to it
+    cache_file = NamedTemporaryFile(mode="w", delete=False)
+    cache_file.write(to_write)
+    cache_file.close()
+
+    # Yield the path to the file
+    yield cache_file.name
+
+    # Delete the file on exit
+    remove(cache_file.name)
+
+
+def test_read_cache_file_not_json():
+    """
+    A cache file that's not JSON should throw a corrupt cache error.
+    """
+    with _read_cache_file_helper("not json") as cache_file_name:
+        with raises(CorruptCacheError):
+            read_cache_file(cache_file_name)
+
+
+def test_read_cache_file_wrong_format():
+    """
+    A cache file with a wrong "__format__" value should throw an assertion error.
+    """
+    with _read_cache_file_helper('{"__format__": 2}') as cache_file_name:
+        with raises(AssertionError):
+            read_cache_file(cache_file_name)
+
+
+def test_read_cache_file_successful():
+    """
+    A good cache file.
+    """
+    with _read_cache_file_helper('{"__format__": 1, "dependencies": "success"}') as cache_file_name:
+        assert "success" == read_cache_file(cache_file_name)
+
+
+def test_reverse_dependencies(from_line):
+    # Since this is a test, make a temporary directory
+    tmp_dir_path = mkdtemp()
```

Comment:
py.test offers a super nice way of creating temp dirs: you can simply have it inject the fixture `tmpdir` to this function, and you can use it by saying `tmp_dir_path = str(tmpdir)`. It'll be cleaned up automatically by pytest.